packages/jitm/src/JITM.php

```diff
@@ -149,6 +153,116 @@ static function jitm_jetpack_woo_services_activate( $CTA ) {
 		);
 	}
 
+	/**
+	 * This is an entire admin notice dedicated to messaging and handling of the case where a user is trying to delete
+	 * the connection owner.
+	 */
+	function delete_user_update_connection_owner_notice() {
+		global $current_screen;
+
+		if ( 'users' !== $current_screen->base ) {
+			return;
+		}
+
+		if ( 'delete' !== $_REQUEST['action'] ) {
+			return;
+		}
+
+		// Get connection owner or bail
+		$connection_manager = new Manager();
+		$connection_owner   = $connection_manager->get_connection_owner();
+		if ( ! $connection_owner ) {
+			return;
+		}
+
+		// Bail if we're not trying to delete connection owner.
+		$user_ids_to_delete        = $_REQUEST['users'];
+		$deleting_connection_owner = in_array( $connection_owner->ID, (array) $user_ids_to_delete );
```

Comment:
Instead of adding `get_connection_owner()` method, would looping `$user_ids_to_delete` using existing `$connection_manager->is_connection_owner()` method be sufficient enough?

The method sounds fairly useful in a connection package tho. :-)

Reply:
What if the site has 10,000 connected users? (I know unlikely, but this is more performant in scale.)

> The method sounds fairly useful in a connection package tho. :-)

I thought so too and was surprised it didn't already exist.